xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubServiceInfo.java

```diff
@@ -117,7 +117,7 @@ public JID getValidJID(String username) {
             try {
                 if (username.contains("@")) {
                     JID jid = new JID(username);
-                    if (userManager.isRegisteredUser(jid)) {
+                    if (userManager.isRegisteredUser(jid, true)) {
```

Comment:
Would this be faster if this came after the check on L123? Do the local search before the remote one?

Reply:
I don't think so. The line on L123 only works if the provided value (confusingly named 'username') is _not_ a JID (or at least, doesn't have an `@` character). It wouldn't work with a JID.

The JID processing bit in `isRegisteredUser` does first check if the JID is local, preventing remote lookups if possible. I'm not seeing how to improve on the execution time significantly.